q2doc/usage/extension.py

```diff
@@ -103,37 +116,48 @@ def run(self):
             if node is not None:
                 nodes_.append(node)
 
-
-        nodes_.append(
+        nodes_.insert(
+            -2,  # bc execution usage should always be _last_
             nodes.literal_block(cmd, cmd, ids=[self._new_id()],
                                 classes=['raw-usage']))
         return nodes_
 
     def setup(self):
         env = self._get_env()
 
-        env.app.q2_plugin_manager = PluginManager()
+        q2_usage = env.app.q2_usage
+        docname = env.docname
+
+        # this means usage-scope directive has not been run in this doc,
+        # so let's use the current docname for the scope's name
+        if docname not in q2_usage['scope_names']:
+            q2_usage['scope_names'][docname] = docname
 
-        if not hasattr(env.app, 'q2_current_doc'):
-            env.app.q2_current_doc = 'base case'
+        scope_name = q2_usage['scope_names'][docname]
 
-        if env.app.q2_current_doc != env.docname:
-            env.app.q2_current_doc = env.docname
+        # save output files based on docname always, rather than using
+        # the scope's name
```

Comment:
Not that it matters, but what was the reasoning for this? It seems like using the scope name has the benefit of behaving the old way when undefined, since it's the document name, and then otherwise, becomes user-defined.

Reply:
Oh I mostly just didn't want to fiddle with the URL-generation in the drivers. I waffled a bit on this, so if you think its important we could just use the scope name, instead of the doc name.

Reply:
Oh that's right, the other consideration is the DEBUG directive for `make preview` on the user docs - that was just another piece I didn't really care to think about, lol.

Reply:
It's not important, if it comes up, it seems like an easy change. And yeah I don't want to think about DEBUG either.